src/AzureAppConfigurationImpl.ts

```diff
@@ -298,9 +376,25 @@ export class AzureAppConfigurationImpl implements AzureAppConfiguration {
         }
     }
 
+    async #refreshFeatureFlags(): Promise<void> {
+        // if still within refresh interval/backoff, return
+        if (!this.#featureFlagRefreshTimer.canRefresh()) {
+            return Promise.resolve();
+        }
+
+        try {
+            await this.#loadFeatureFlags();

```

Comment:
we should check if any feature flags changed before loading all feature flags again. I think collection monitoring API can be leveraged here.

Reply:
Agree. Ultimately we should leverage the collection monitoring api. I prefer we simply poll all ffs here for simplicity. I’ll check if it’s ready in latest js sdk and impl that in a separate PR.

Reply:
This could lead to a lot of server requests if someone is using a short refresh interval. It's almost like a forced refresh of feature flags whenever refresh interval elapses. This is conceptually very different from what other providers are doing. In other providers, we check the etag of each feature flag and reload feature flags only if at least 1 flag has changed on the server. Here's the Python implementation for reference: https://github.com/Azure/azure-sdk-for-python/blob/e08b3b0f8843a38407515b91a72638aa3dd09447/sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py#L584-L639

If the new Page etag API is close to being released, I would say lets implement that directly instead of tracking individual etags ourselves. Other providers were released before we had the API, so they implment the workaround of tracking etags manually. But either way, we should not reload feature flags if nothing has changed on the server.

Reply:
It's already released in js-sdk v1.6.0. Sounds reasonable, we should make sure it's correctly implemented before we ship it... I'll work on the corresponding impl.

Reply:
update sdk version to 1.6 in #76 . and we update this PR to use page etag api here then.